plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadataConfig.java

```diff
@@ -57,6 +58,19 @@ public JdbcMetadataConfig setAggregationPushdownEnabled(boolean aggregationPushd
         return this;
     }
 
+    public boolean isCaseInsensitivePredicatePushdownEnabled()
+    {
+        return caseInsensitivePredicatePushdownEnabled;
+    }
+
+    @Config("case-insensitive-predicate-pushdown.enabled")
```

Comment:
Consider the case where someone sets their MySQL instance to use case sensitive collation by default and they want to leverage the full pushdown capabilities. In such a situation, the "insensitive" in the flag name is not reasonable, as this is exactly what they are _not_ doing.

Maybe `unsafe-varchar-predicate-pushdown.enabled` ?

cc @losipiuk @electrum

Reply:
Hm... in case of PostgreSQL or other case sensitive database one case change collation to use case insensitive comparisons. Then we might need to do something reverse to match the remote database collation.

How about `varchar-predicate-pushdown.mode`  with values `FULL`, `PUSHDOWN_AND_KEEP` and `CASE_INSENSITIVE`? Then depending on connector we could have different defaults. WDYT?

Reply:
That's an option, but IMO exposes too much internal detail, while being also pretty technical and I am not yet convinced it's the right way.
Also, why would we have the toggle for a connector which doesn't need it, because remote has just one collation?